### ISSUE TITLE
[FR]: Support custom providers with AuthProviderID

### ISSUE THREAD

**Issue by kazumanagano** [firebase/firebase-ios-sdk#13429]
### Description

Since #13097, `OAuthProvidel.credential` method now accepts the AuthProviderID as an Enum.

In addition, methods that have the existing String type providerID as an argument now display a Deprecated warning and request migration.

However, this does not support custom ProviderIDs such as OpenID Connect, so products that use it cannot be migrated.
https://firebase.google.com/docs/auth/ios/openid-connect#manual

### API Proposal

Please make it possible to take a custom ProviderID as an argument as shown below, or prepare a separate method.

```
public enum AuthProviderID: String {
  case apple = "apple.com"
  case email = "password"
  case facebook = "facebook.com"
  case gameCenter = "gc.apple.com"
  case gitHub = "github.com"
  case google = "google.com"
  case phone
  case custom(String)
}
```

OR

```
  // For Custom Provider
  public static func credential(providerID: String, idToken: String,
                                rawNonce: String,
                                accessToken: String? = nil) -> OAuthCredential {
```

### Firebase Product(s)

Authentication

**Comment by paulb777** [comment]
@kazumanagano Thanks for the report. We appreciate your quick feedback on 11.0.0!

I've proposed a fix at #13433 and will follow up with the team on getting it reviewed and merged.

**Comment by kazumanagano** [comment]
Thank you for your great and quick fix! 
I'm looking forward to the release!
